hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/recordcache/RecordCacheImpl.java

```diff
@@ -178,8 +182,9 @@ public void add(
 
         // To avoid having a background thread cleaning out this queue, we spend a little time when adding to the queue
         // to also remove from the queue any transactions that have expired.
-        final var queue = getQueue();
-        final var firstRecord = transactionRecords.get(0);
+        final WritableStates states = getWritableState();
+        final WritableQueueState<TransactionRecordEntry> queue = states.getQueue(TXN_RECORD_QUEUE);
+        final SingleTransactionRecord firstRecord = transactionRecords.get(0);
```

Comment:
Nit: is the queue guaranteed to always have at least one record?

Reply:
I believe that at this point the queue state will always return something from get(0), as it was tested non-empty before getting here (line 178).

Reply:
Missed that, thanks